File: src/CSharp/801-900/872.LeafSimilarTrees.cs
using System;
using System.Collections.Generic;

namespace PeefyLeetCode.LeafSimilar
{
    public class TreeNode
    {
        public int val;
        public TreeNode left;
        public TreeNode right;
        public TreeNode(int x) { val = x; }
    }

public class Solution {
    private void order(TreeNode node, List<int> list){
        if (node == null)
            return;
        if (node.left == null && node.right == null){
            list.Add(node.val);
            return;
        }
        order(node.left, list);
        order(node.right, list);
    }
    public bool LeafSimilar(TreeNode root1, TreeNode root2) {
        List<int> list1 = new List<int>();
        List<int> list2 = new List<int>();
        order(root1, list1);
        order(root2, list2);
        if (list1.Count != list2.Count)
            return false;
        for(int i = 0; i < list1.Count; i++){
            if (list1[i] != list2[i])
                return false;
        }
        return true;
    }
}

}

File: src/Python/801-900/872.LeafSimilarTrees.py

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def getlist(self, root):
        self.l = []
        def order(node):
            if node is None:
                return
            if node.left is None and node.right is None:
                self.l.append(node.val)
                return
            order(node.left)
            order(node.right)
        order(root)
        return self.l

    def leafSimilar(self, root1, root2):
        """
        :type root1: TreeNode
        :type root2: TreeNode
        :rtype: bool
        """
        return self.getlist(root1) == self.getlist(root2)


if __name__ == '__main__':
    solution = Solution()
    root1 = TreeNode(1)
    root1.left = TreeNode(2)
    root2 = TreeNode(1)
    root2.right = TreeNode(2)
    print(solution.leafSimilar(root1, root2))

else:
    pass
